### ISSUE TITLE
readAxivity Changed?

### ISSUE THREAD

**Issue by muschellij2** [wadpac/GGIRread#67]
Do you know what causes this change in the data output?
I see some comments 
https://github.com/wadpac/GGIRread/blob/main/inst/NEWS.Rd#L40
https://github.com/wadpac/GGIRread/blob/main/inst/NEWS.Rd#L58
But I'm unsure.

Are there tests to ensure the same output from version to version?

```r
> remotes::install_version("GGIRread", version = "0.2.0")
Downloading package from url: https://cran.rstudio.com//src/contrib/Archive/GGIRread/GGIRread_0.2.0.tar.gz
Loading required package: colorout
* installing *source* package ‘GGIRread’ ...
** package ‘GGIRread’ successfully unpacked and MD5 sums checked
** using staged installation
** libs
using C compiler: ‘Apple clang version 15.0.0 (clang-1500.3.9.4)’
using C++ compiler: ‘Apple clang version 15.0.0 (clang-1500.3.9.4)’
using SDK: ‘MacOSX14.4.sdk’
clang++ -mmacosx-version-min=10.13 -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/Rcpp/include' -I/opt/R/x86_64/include    -fPIC  -stdlib=libc++  -c AxivityNumUnpack.cpp -o AxivityNumUnpack.o
clang++ -mmacosx-version-min=10.13 -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/Rcpp/include' -I/opt/R/x86_64/include    -fPIC  -stdlib=libc++  -c GENEActivReader.cpp -o GENEActivReader.o
clang++ -mmacosx-version-min=10.13 -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/Rcpp/include' -I/opt/R/x86_64/include    -fPIC  -stdlib=libc++  -c RcppExports.cpp -o RcppExports.o
clang -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/Rcpp/include' -I/opt/R/x86_64/include    -fPIC  -falign-functions=64 -Wall -g -O2  -c init.c -o init.o
clang++ -mmacosx-version-min=10.13 -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/Rcpp/include' -I/opt/R/x86_64/include    -fPIC  -stdlib=libc++  -c resample.cpp -o resample.o
clang++ -mmacosx-version-min=10.13 -std=gnu++11 -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -L/Library/Frameworks/R.framework/Resources/lib -stdlib=libc++ -o GGIRread.so AxivityNumUnpack.o GENEActivReader.o RcppExports.o init.o resample.o -F/Library/Frameworks/R.framework/.. -framework R -Wl,-framework -Wl,CoreFoundation
ld: warning: object file (/private/var/folders/1s/wrtqcpxn685_zk570bnx9_rr0000gr/T/RtmpUTGCtZ/remotes141b34d45b00c/GGIRread/src/init.o) was built for newer 'macOS' version (14.0) than being linked (10.13)
installing to /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/00LOCK-GGIRread/00new/GGIRread/libs
** R
** inst
** byte-compile and prepare package for lazy loading
Loading required package: colorout
** help
*** installing help indices
** building package indices
Loading required package: colorout
** testing if installed package can be loaded from temporary location
Loading required package: colorout
** checking absolute paths in shared objects and dynamic libraries
** testing if installed package can be loaded from final location
Loading required package: colorout
** testing if installed package keeps a record of temporary installation path
* DONE (GGIRread)
> sessioninfo::package_info("GGIRread")
 package  * version date (UTC) lib source
 bitops     1.0-7   2021-04-24 [1] CRAN (R 4.4.0)
 GGIRread   0.2.0   2022-08-15 [1] CRAN (R 4.4.0)
 matlab     1.0.4   2022-06-01 [1] CRAN (R 4.4.0)
 Rcpp       1.0.12  2024-01-09 [1] CRAN (R 4.4.0)

 [1] /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library
> file = system.file("testfiles", "ax3_testfile.cwa", package = "GGIRread")
> res = GGIRread::readAxivity(file, desiredtz = "UTC",
+                             start = 0,
+                             end = Inf,
+                             progressBar = TRUE)
  |================================================================================================                       |  80%> dput(mean(res$data$x))
> dput(head(res$data$x))
0.775495573675064
c(0.328125, 0.822095085973384, 0.873869391120009, 0.890059695560005, 
0.890625, 0.875942174066659)

Restarting R session...
```

```r
> install.packages("GGIRread")
trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-x86_64/contrib/4.4/GGIRread_1.0.1.tgz'
Content type 'application/x-gzip' length 879657 bytes (859 KB)
==================================================
downloaded 859 KB


The downloaded binary packages are in
	/var/folders/1s/wrtqcpxn685_zk570bnx9_rr0000gr/T//Rtmpo7861y/downloaded_packages
> sessioninfo::package_info("GGIRread")
 package  * version date (UTC) lib source
 bitops     1.0-7   2021-04-24 [1] CRAN (R 4.4.0)
 **GGIRread   1.0.1   2024-06-03 [1] CRAN (R 4.4.0)**
 matlab     1.0.4   2022-06-01 [1] CRAN (R 4.4.0)
 Rcpp       1.0.12  2024-01-09 [1] CRAN (R 4.4.0)

 [1] /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library
> file = system.file("testfiles", "ax3_testfile.cwa", package = "GGIRread")
> res = GGIRread::readAxivity(file, desiredtz = "UTC",
+                             start = 0,
+                             end = Inf,
+                             progressBar = TRUE)
  |=====================================================================================================================  |  99%
> dput(mean(res$data$x))
0.775495066145421
> dput(head(res$data$x))
c(0.328125, 0.822095085973384, 0.873869391120009, 0.890059695560005, 
0.890625, 0.875942174066659)
```

# Direct Comparison
```r

remotes::install_version("GGIRread", version = "0.2.0")
sessioninfo::package_info("GGIRread")
file = system.file("testfiles", "ax3_testfile.cwa", package = "GGIRread")
res = GGIRread::readAxivity(file, desiredtz = "UTC",
                            start = 0,
                            end = Inf,
                            progressBar = TRUE)
dput(mean(res$data$x))
dput(head(res$data$x))
saveRDS(res, file = "GGIR_read_0.2.0_ax3_output.rds")
```
( Restart R )
```r
install.packages("GGIRread")
sessioninfo::package_info("GGIRread")
file = system.file("testfiles", "ax3_testfile.cwa", package = "GGIRread")
res = GGIRread::readAxivity(file, desiredtz = "UTC",
                            start = 0,
                            end = Inf,
                            progressBar = TRUE)
dput(mean(res$data$x))
dput(head(res$data$x))
old_res = readRDS(file = "GGIR_read_0.2.0_ax3_output.rds")
all.equal(res, old_res)
```

```r
> all.equal(res, old_res)
[1] "Length mismatch: comparison on first 2 components"                        
[2] "Component “header”: Length mismatch: comparison on first 8 components"    
[3] "Component “data”: Component “x”: Mean relative difference: 0.007375119"   
[4] "Component “data”: Component “y”: Mean relative difference: 0.0303068"     
[5] "Component “data”: Component “z”: Mean relative difference: 0.01141777"    
[6] "Component “data”: Component “temp”: Mean relative difference: 0.2944787"  
[7] "Component “data”: Component “battery”: Mean absolute difference: 4.113403"
```

**Comment by muschellij2** [comment]
It looks as though it was between 0.2.0 and 0.3.1

```
> remotes::install_version("GGIRread", version = "0.3.1")
Downloading package from url: https://cran.rstudio.com//src/contrib/Archive/GGIRread/GGIRread_0.3.1.tar.gz
Loading required package: colorout
* installing *source* package ‘GGIRread’ ...
** package ‘GGIRread’ successfully unpacked and MD5 sums checked
** using staged installation
** libs
using C compiler: ‘Apple clang version 15.0.0 (clang-1500.3.9.4)’
using C++ compiler: ‘Apple clang version 15.0.0 (clang-1500.3.9.4)’
using SDK: ‘MacOSX14.4.sdk’
clang++ -mmacosx-version-min=10.13 -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/Rcpp/include' -I/opt/R/x86_64/include    -fPIC  -stdlib=libc++  -c AxivityNumUnpack.cpp -o AxivityNumUnpack.o
clang++ -mmacosx-version-min=10.13 -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/Rcpp/include' -I/opt/R/x86_64/include    -fPIC  -stdlib=libc++  -c GENEActivReader.cpp -o GENEActivReader.o
clang++ -mmacosx-version-min=10.13 -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/Rcpp/include' -I/opt/R/x86_64/include    -fPIC  -stdlib=libc++  -c RcppExports.cpp -o RcppExports.o
clang -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/Rcpp/include' -I/opt/R/x86_64/include    -fPIC  -falign-functions=64 -Wall -g -O2  -c init.c -o init.o
clang++ -mmacosx-version-min=10.13 -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/Rcpp/include' -I/opt/R/x86_64/include    -fPIC  -stdlib=libc++  -c resample.cpp -o resample.o
clang++ -mmacosx-version-min=10.13 -std=gnu++11 -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -L/Library/Frameworks/R.framework/Resources/lib -stdlib=libc++ -o GGIRread.so AxivityNumUnpack.o GENEActivReader.o RcppExports.o init.o resample.o -F/Library/Frameworks/R.framework/.. -framework R -Wl,-framework -Wl,CoreFoundation
ld: warning: object file (/private/var/folders/1s/wrtqcpxn685_zk570bnx9_rr0000gr/T/Rtmp24huum/remotes1481a74f67e5c/GGIRread/src/init.o) was built for newer 'macOS' version (14.0) than being linked (10.13)
installing to /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/00LOCK-GGIRread/00new/GGIRread/libs
** R
** inst
** byte-compile and prepare package for lazy loading
Loading required package: colorout
** help
*** installing help indices
** building package indices
Loading required package: colorout
** testing if installed package can be loaded from temporary location
Loading required package: colorout
** checking absolute paths in shared objects and dynamic libraries
** testing if installed package can be loaded from final location
Loading required package: colorout
** testing if installed package keeps a record of temporary installation path
* DONE (GGIRread)
> sessioninfo::package_info("GGIRread")
 package  * version date (UTC) lib source
 bitops     1.0-7   2021-04-24 [1] CRAN (R 4.4.0)
 GGIRread   0.3.1   2023-10-11 [1] CRAN (R 4.4.0)
 matlab     1.0.4   2022-06-01 [1] CRAN (R 4.4.0)
 Rcpp       1.0.12  2024-01-09 [1] CRAN (R 4.4.0)

 [1] /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library
> file = system.file("testfiles", "ax3_testfile.cwa", package = "GGIRread")
> res = GGIRread::readAxivity(file, desiredtz = "UTC",
+                             start = 0,
+                             end = Inf,
+                             progressBar = FALSE)
> dput(mean(res$data$x))
0.775495066145421
> dput(head(res$data$x))
c(0.328125, 0.822095085973384, 0.873869391120009, 0.890059695560005, 
0.890625, 0.875942174066659)
> saveRDS(res, file = "GGIR_read_0.3.1_ax3_output.rds")
```
Restarting R session...
```r
> install.packages("GGIRread")
trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-x86_64/contrib/4.4/GGIRread_1.0.1.tgz'
Content type 'application/x-gzip' length 879657 bytes (859 KB)
==================================================
downloaded 859 KB


The downloaded binary packages are in
	/var/folders/1s/wrtqcpxn685_zk570bnx9_rr0000gr/T//Rtmp2jPNkt/downloaded_packages
> sessioninfo::package_info("GGIRread")
 package  * version date (UTC) lib source
 bitops     1.0-7   2021-04-24 [1] CRAN (R 4.4.0)
 GGIRread   1.0.1   2024-06-03 [1] CRAN (R 4.4.0)
 matlab     1.0.4   2022-06-01 [1] CRAN (R 4.4.0)
 Rcpp       1.0.12  2024-01-09 [1] CRAN (R 4.4.0)

 [1] /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library
> file = system.file("testfiles", "ax3_testfile.cwa", package = "GGIRread")
> res = GGIRread::readAxivity(file, desiredtz = "UTC",
+                             start = 0,
+                             end = Inf,
+                             progressBar = TRUE)
  |=====================================================================================================================  |  99%> dput(mean(res$data$x))
0.775495066145421
> dput(head(res$data$x))
c(0.328125, 0.822095085973384, 0.873869391120009, 0.890059695560005, 
0.890625, 0.875942174066659)
> old_res = readRDS(file = "GGIR_read_0.3.1_ax3_output.rds")
> all.equal(res, old_res)
[1] TRUE
```

**Comment by muschellij2** [comment]
Happened between 0.2.8 and 0.3.0
``` r
# remotes::install_version("GGIRread", version = "0.2.8")
sessioninfo::package_info("GGIRread")
#>  package  * version  date (UTC) lib source
#>  bitops     1.0-7    2021-04-24 [1] CRAN (R 4.4.0)
#>  GGIRread   0.2.8    2023-05-26 [1] CRAN (R 4.4.0)
#>  MASS       7.3-60.2 2024-04-24 [1] local
#>  matlab     1.0.4    2022-06-01 [1] CRAN (R 4.4.0)
#>  Rcpp       1.0.12   2024-01-09 [1] CRAN (R 4.4.0)
#>  signal     1.8-0    2023-11-27 [1] CRAN (R 4.4.0)
#>  tuneR      1.4.7    2024-04-16 [1] CRAN (R 4.4.0)
#> 
#>  [1] /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library
```

``` r
file = system.file("testfiles", "ax3_testfile.cwa", package = "GGIRread")
res = GGIRread::readAxivity(file, desiredtz = "UTC",
                            start = 0,
                            end = Inf,
                            progressBar = FALSE)
dput(mean(res$data$x))
#> 0.775495573675064
```

``` r
dput(head(res$data$x))
#> c(0.328125, 0.822095085973384, 0.873869391120009, 0.890059695560005, 
#> 0.890625, 0.875942174066659)
```

``` r
new_res = readRDS(file = "GGIR_read_ax3_output.rds")
#> Warning in gzfile(file, "rb"): cannot open compressed file
#> 'GGIR_read_ax3_output.rds', probable reason 'No such file or directory'
#> Error in gzfile(file, "rb"): cannot open the connection
```

``` r
all.equal(res, new_res)
#> Error in eval(expr, envir, enclos): object 'new_res' not found
```

<sup>Created on 2024-06-06 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.4.0 (2024-04-24)
#>  os       macOS Sonoma 14.4.1
#>  system   x86_64, darwin20
#>  ui       X11
#>  language (EN)
#>  collate  en_US.UTF-8
#>  ctype    en_US.UTF-8
#>  tz       America/New_York
#>  date     2024-06-06
#>  pandoc   3.1.11.1 @ /usr/local/bin/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version date (UTC) lib source
#>  cli           3.6.2   2023-12-11 [1] CRAN (R 4.4.0)
#>  digest        0.6.35  2024-03-11 [1] CRAN (R 4.4.0)
#>  evaluate      0.23    2023-11-01 [1] CRAN (R 4.4.0)
#>  fastmap       1.2.0   2024-05-15 [1] CRAN (R 4.4.0)
#>  fs            1.6.4   2024-04-25 [1] CRAN (R 4.4.0)
#>  GGIRread      0.2.8   2023-05-26 [1] CRAN (R 4.4.0)
#>  glue          1.7.0   2024-01-09 [1] CRAN (R 4.4.0)
#>  htmltools     0.5.8.1 2024-04-04 [1] CRAN (R 4.4.0)
#>  knitr         1.46    2024-04-06 [1] CRAN (R 4.4.0)
#>  lifecycle     1.0.4   2023-11-07 [1] CRAN (R 4.4.0)
#>  magrittr      2.0.3   2022-03-30 [1] CRAN (R 4.4.0)
#>  purrr         1.0.2   2023-08-10 [1] CRAN (R 4.4.0)
#>  R.cache       0.16.0  2022-07-21 [1] CRAN (R 4.4.0)
#>  R.methodsS3   1.8.2   2022-06-13 [1] CRAN (R 4.4.0)
#>  R.oo          1.26.0  2024-01-24 [1] CRAN (R 4.4.0)
#>  R.utils       2.12.3  2023-11-18 [1] CRAN (R 4.4.0)
#>  Rcpp          1.0.12  2024-01-09 [1] CRAN (R 4.4.0)
#>  reprex        2.1.0   2024-01-11 [1] CRAN (R 4.4.0)
#>  rlang         1.1.3   2024-01-10 [1] CRAN (R 4.4.0)
#>  rmarkdown     2.27    2024-05-17 [1] CRAN (R 4.4.0)
#>  rstudioapi    0.16.0  2024-03-24 [1] CRAN (R 4.4.0)
#>  sessioninfo   1.2.2   2021-12-06 [1] CRAN (R 4.4.0)
#>  styler        1.10.3  2024-04-07 [1] CRAN (R 4.4.0)
#>  vctrs         0.6.5   2023-12-01 [1] CRAN (R 4.4.0)
#>  withr         3.0.0   2024-01-16 [1] CRAN (R 4.4.0)
#>  xfun          0.44    2024-05-15 [1] CRAN (R 4.4.0)
#>  yaml          2.3.8   2023-12-11 [1] CRAN (R 4.4.0)
#> 
#>  [1] /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

**Comment by muschellij2** [comment]
Still present at `0.2.10` compared to current version.

``` r
# remotes::install_github("wadpac/GGIRread@dc6466eec76a90667bac548374c28fe831270b2a")
sessioninfo::package_info("GGIRread")
#>  package  * version  date (UTC) lib source
#>  bitops     1.0-7    2021-04-24 [1] CRAN (R 4.4.0)
#>  GGIRread   0.2.10   2024-06-06 [1] Github (wadpac/GGIRread@dc6466e)
#>  MASS       7.3-60.2 2024-04-24 [1] local
#>  matlab     1.0.4    2022-06-01 [1] CRAN (R 4.4.0)
#>  Rcpp       1.0.12   2024-01-09 [1] CRAN (R 4.4.0)
#>  signal     1.8-0    2023-11-27 [1] CRAN (R 4.4.0)
#>  tuneR      1.4.7    2024-04-16 [1] CRAN (R 4.4.0)
#> 
#>  [1] /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library
```

``` r
file = system.file("testfiles", "ax3_testfile.cwa", package = "GGIRread")
res = GGIRread::readAxivity(file, desiredtz = "UTC",
                            start = 0,
                            end = Inf,
                            progressBar = FALSE)
dput(mean(res$data$x))
#> 0.775495573675064
```

``` r
dput(head(res$data$x))
#> c(0.328125, 0.822095085973384, 0.873869391120009, 0.890059695560005, 
#> 0.890625, 0.875942174066659)
```

``` r
new_res = readRDS(file = "GGIR_read_ax3_output.rds")
#> Warning in gzfile(file, "rb"): cannot open compressed file
#> 'GGIR_read_ax3_output.rds', probable reason 'No such file or directory'
#> Error in gzfile(file, "rb"): cannot open the connection
```

``` r
all.equal(res, new_res)
#> Error in eval(expr, envir, enclos): object 'new_res' not found
```

<sup>Created on 2024-06-06 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.4.0 (2024-04-24)
#>  os       macOS Sonoma 14.4.1
#>  system   x86_64, darwin20
#>  ui       X11
#>  language (EN)
#>  collate  en_US.UTF-8
#>  ctype    en_US.UTF-8
#>  tz       America/New_York
#>  date     2024-06-06
#>  pandoc   3.1.11.1 @ /usr/local/bin/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version date (UTC) lib source
#>  cli           3.6.2   2023-12-11 [1] CRAN (R 4.4.0)
#>  digest        0.6.35  2024-03-11 [1] CRAN (R 4.4.0)
#>  evaluate      0.23    2023-11-01 [1] CRAN (R 4.4.0)
#>  fastmap       1.2.0   2024-05-15 [1] CRAN (R 4.4.0)
#>  fs            1.6.4   2024-04-25 [1] CRAN (R 4.4.0)
#>  GGIRread      0.2.10  2024-06-06 [1] Github (wadpac/GGIRread@dc6466e)
#>  glue          1.7.0   2024-01-09 [1] CRAN (R 4.4.0)
#>  htmltools     0.5.8.1 2024-04-04 [1] CRAN (R 4.4.0)
#>  knitr         1.46    2024-04-06 [1] CRAN (R 4.4.0)
#>  lifecycle     1.0.4   2023-11-07 [1] CRAN (R 4.4.0)
#>  magrittr      2.0.3   2022-03-30 [1] CRAN (R 4.4.0)
#>  purrr         1.0.2   2023-08-10 [1] CRAN (R 4.4.0)
#>  R.cache       0.16.0  2022-07-21 [1] CRAN (R 4.4.0)
#>  R.methodsS3   1.8.2   2022-06-13 [1] CRAN (R 4.4.0)
#>  R.oo          1.26.0  2024-01-24 [1] CRAN (R 4.4.0)
#>  R.utils       2.12.3  2023-11-18 [1] CRAN (R 4.4.0)
#>  Rcpp          1.0.12  2024-01-09 [1] CRAN (R 4.4.0)
#>  reprex        2.1.0   2024-01-11 [1] CRAN (R 4.4.0)
#>  rlang         1.1.3   2024-01-10 [1] CRAN (R 4.4.0)
#>  rmarkdown     2.27    2024-05-17 [1] CRAN (R 4.4.0)
#>  rstudioapi    0.16.0  2024-03-24 [1] CRAN (R 4.4.0)
#>  sessioninfo   1.2.2   2021-12-06 [1] CRAN (R 4.4.0)
#>  styler        1.10.3  2024-04-07 [1] CRAN (R 4.4.0)
#>  vctrs         0.6.5   2023-12-01 [1] CRAN (R 4.4.0)
#>  withr         3.0.0   2024-01-16 [1] CRAN (R 4.4.0)
#>  xfun          0.44    2024-05-15 [1] CRAN (R 4.4.0)
#>  yaml          2.3.8   2023-12-11 [1] CRAN (R 4.4.0)
#> 
#>  [1] /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

**Comment by vincentvanhees** [comment]
Yes, `readAxivity` has changed a lot in 2023 where consistency breaking changes were made.

The very first version of the code was written by a contributor, it seemed to work but little effort was made to review it. As of release 0.2.7 (May 2023) I started reviewing and improving the code. As of release 0.2.10 (August 2023) Lena became involved resulting in further improvements in this and the following releases. As of 0.3.2 (Dec 2023) no additional changes were made and the unit test should help to preserve consistency moving forward. We think this version of the code is as good as it can get. Collaborators have ran it in UK Biobank without any noticeable problems so far.

**Comment by muschellij2** [comment]
I would suggest changing the major version as this is a breaking change.

**Comment by vincentvanhees** [comment]
I did change it form 0. to 1. after no additional bugs or problems were found in the months after 0.3.2. During the whole process it did not seem desirable to create a new major release after each enhancement. The situation was that the code was poor to start with and I knew that it would take several iterations to get to a more acceptable version.

The updates to readGENEActiv on the contrary were relatively minor once a first version had been released. Probably because we made sure that the code was good before its first release.
